enterprise_subsidy/apps/api/v1/views/subsidy.py

```diff
@@ -174,3 +197,78 @@ def can_redeem(self, request, uuid):  # pylint: disable=unused-argument
             )
         )
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+    @extend_schema(
+        tags=['subsidy'],
+        request=SubsidyCreationRequestSerializer,
+        responses={
+            201: SubsidySerializer,
+            200: SubsidySerializer,
+            403: exceptions.PermissionDenied,
+            400: exceptions.ValidationError,
+            500: exceptions.APIException,
+        },
+    )
+    def create(self, request, *args, **kwargs):
+        """
+        Get or create a new subsidy
+
+        Endpoint Location: POST /api/v1/subsidies/
+        """
+        create_serializer = SubsidyCreationRequestSerializer(data=request.data)
+        if create_serializer.is_valid(raise_exception=True):
+            try:
+                subsidy, created = get_or_create_learner_credit_subsidy(
+                    create_serializer.data['reference_id'],
+                    create_serializer.data['default_title'],
+                    create_serializer.data['default_enterprise_customer_uuid'],
+                    create_serializer.data['default_unit'],
+                    create_serializer.data['default_starting_balance'],
+                    create_serializer.data['default_revenue_category'],
+                    create_serializer.data['default_internal_only'],
+                )
+                if created:
+                    return Response(SubsidySerializer(subsidy).data, status=status.HTTP_201_CREATED)
+                if subsidy:
+                    return Response(SubsidySerializer(subsidy).data, status=status.HTTP_200_OK)
+                else:
+                    assert False
```

Comment:
You might also want to raise a 500 here, too - it'd be a truly "exceptional" condition if attempting to get or create returned nothing.